firebase-common/src/main/java/com/google/firebase/concurrent/CustomThreadFactory.java

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.concurrent;
+
+import java.util.Locale;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicLong;
+
+class CustomThreadFactory implements ThreadFactory {
+  private static final ThreadFactory DEFAULT = Executors.defaultThreadFactory();
+  private final AtomicLong threadCount = new AtomicLong();
```

Comment:
why the counter?

Reply:
This is common practice to give different names to threads, i.e. `executor-foo-1`, `executor-foo-2` etc.

Reply:
Didn't see it in the name, thanks!